infrastructure-playbooks/rolling_update.yml

```diff
@@ -536,25 +536,83 @@
         - (ceph_versions.get('stdout', '{}')|from_json).get('osd', {}) | length == 1
         - ceph_versions_osd | string is search("ceph version 12")
 
-- name: upgrade ceph mdss cluster
-  vars:
-    upgrade_ceph_packages: True
-  hosts: "{{ mds_group_name|default('mdss') }}"
-  serial: 1
-  become: True
+
+- name: upgrade ceph mdss cluster, deactivate all rank > 0
+  hosts: "{{ groups[mon_group_name|default('mons')][0] }}"
+  become: true
   tasks:
-    - name: stop ceph mds
+    - import_role:
+        name: ceph-defaults
+    - import_role:
+        name: ceph-facts
+
+    - name: set max_mds 1 on ceph fs
+      command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs set {{ cephfs }} max_mds 1"
+      changed_when: false
+
+    - name: wait until only rank 0 is up
+      command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs get {{ cephfs }} -f json"
+      changed_when: false
+      register: wait_rank_zero
+      retries: 200
```

Comment:
not sure about what would be good values here for retries/delay

Reply:
@batrick any idea?

Reply:
This can potentially take a long time to finish. It depends on the number of ranks and the file system load. What happens if the timeout expires?

I would suggest a delay of 5 seconds and retries of 720; or 1 hour of retries.